docs/docs/when-to-use-use.md

```diff
@@ -0,0 +1,232 @@
+# When to use `use`?
+
+`use` connects resources to _JavaScript_ contexts.
+
+Note that all of these examples use a `Component` class, but they could just as well be any native vanilla JavaScript class. See [link](https://ember-resources.pages.dev/funcs/link.link) for more on that.
+
+## `import { use } from 'ember-resources';`
+
+This type of use is for connecting a resource to a JS class.
+
+It has a couple different overloads so that the community can provide feedback on the ergonomics of each over time.
+
+
+<details><summary>Under the hood</summary>
+
+`use` uses [`invokeHelper`](https://api.emberjs.com/ember/release/functions/@ember%2Fhelper/invokeHelper) which means that resources _are_ helpers. It also means that other non-resource helpers could be passed to `use` (note tho that non-resource usage is not supported by this library).
+
+There is a potential for a whole "usable system" here, piggy-backing off of the [helper-manager](https://rfcs.emberjs.com/id/0625-helper-managers/) pattern.
+
+</details>
+
+### `@use`
+
+
+#### "@use thisProperty = DefinedAsThis"
+
+This `@use` decorator _replaces_ the property with a getter that gets the underlying value of the resource. You'll see what happens in non-decorator form later.
+
+<details><summary>looking under the hood</summary>
+
+You could think of what the decorator does in this way:
+
+```js
+class Demo extends Component {
+  @use data = Clock;
+}
+```
+is (at runtime) transformed to:
+```js
+class Demo extends Component {
+  #data1 = invokeHelper(this, Clock);
+  
+  get data() {
+    return getValue(this.#data1);
+  }
+}
+```
+
+Note that the _actual_ return value of `resource` is an internal and private object, and interaction with that object is not supported in user-space.
+
+</details>
+
+In this example, the resource is the value of `data`:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource(() => {
+    /* ... */
+    return 2;
+});
+
+class Demo extends Component {
+    @use data = Clock;
+
+    <template>
+        {{this.time}} prints 2
```

Comment:
Where's `this.time` come from? Should it be `this.data`?

Reply:
thanks! fixed!